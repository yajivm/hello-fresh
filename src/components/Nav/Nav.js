import { useState } from 'react';
import PropTypes from 'prop-types';
import NavItem from "./NavItem";
import { NavWrapper } from './Nav.styles';

const Nav = ({ navList }) => {
  const [activeTab, setActiveTab] = useState(navList[0].title);

  const setActiveTabStatus = title => {
    setActiveTab(title);
  };

  return (
    <NavWrapper>
      {navList.map((item, ind) => (
        <NavItem title={item.title} key={ind} to={item.path} onClick={setActiveTabStatus} isCurrentTabActive={activeTab === item.title}/>
      ))}
    </NavWrapper>
  )
}

Nav.propTypes = {
  navList: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default Nav;