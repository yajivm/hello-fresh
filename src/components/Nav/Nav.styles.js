import styled from "styled-components";

export const NavWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  background: #fff;
  width: 100%;
  display: flex;
  justify-content: center;
  box-shadow: rgba(0,0,0,0.2) 0px 0px 8px 0px;
`