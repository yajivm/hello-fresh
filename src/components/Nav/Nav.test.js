import React from 'react'
import { render } from '@testing-library/react'
import Nav from '.'

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');
  return {
    ...originalModule,
    Link: props => <div {...props}></div>,
  };
});

const props = {
  navList: [
    {
      title: 'Test',
      path: '/',
    }
  ]
}

describe('Nav', () => {

  it('should render component', () => {
    const { container } = render(<Nav {...props} />)

    expect(container).toMatchSnapshot();
  })
})