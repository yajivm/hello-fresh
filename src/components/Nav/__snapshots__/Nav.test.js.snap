// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render component 1`] = `
<div>
  <div
    class="sc-gsnTZi bLzQqw"
  >
    <div
      class="sc-bczRLJ fEEeGt"
      data-testid="Test"
      to="/"
    >
      Test
    </div>
  </div>
</div>
`;
